packages/jupyter-ai/jupyter_ai/chat_handlers/export.py

```diff
@@ -25,21 +26,23 @@ def chat_message_to_markdown(self, message):
         else:
             return ""
 
-    # Multiple chat histories in separate files
-    def get_chat_filename(self, path="./chat_history.md"):
-        filename, extension = os.path.splitext(path)
-        counter = 1
-        while os.path.exists(path):
-            path = filename + "_" + str(counter) + ".md"
-            counter += 1
+    # Multiple chat histories can be saved in separate files, each with a timestamp
+    def get_chat_filename(self, file_id):
+        file_id = file_id.split(" ")
+        if len(file_id) > 1:
+            file_id = file_id[-1]
+        else:
+            file_id = "chat_history"
+        outfile = f"{file_id}-{datetime.now():%Y-%m-%d-%H-%M}.md"
+        path = os.path.join(self.root_dir, outfile)
         return path
 
     async def process_message(self, _):
         markdown_content = "\n\n".join(
             self.chat_message_to_markdown(msg) for msg in self._chat_history
         )
         # Write the markdown content to a file or do whatever you want with it
-        chat_filename = self.get_chat_filename()
+        chat_filename = self.get_chat_filename(self._chat_history[-1].body)
```

Comment:
Using chat_history for the user message doesn't seem correct here. Look at the [learn](https://github.com/jupyterlab/jupyter-ai/blob/main/packages/jupyter-ai/jupyter_ai/chat_handlers/learn.py#L40) handler how it uses argparse to parse the user input. We only need the prog, and the path.
```python
self.parser.prog = "/export"
self.parser.add_argument("path", nargs=argparse.REMAINDER)
```

Reply:
Suggested approach has been implemented and `argparse` is now used to get the output file name if used.